### PR TITLE
feat: introduce tasklist-task template in new exporter

### PR DESCRIPTION
**Pull request by panagiotisgts**
## Description

Adapt previous `tasklist-task` index template to match the structure of the old `tasklist-list-view` template(definition in the ref issue).  Include `tasklist-task` index template in CamundaExporter schema.

Part of: #23040
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: webapps-schema/src/main/java/io/camunda/webapps/schema/entities/tasklist/TasklistEntity.java [hunk 1]
```diff
@@ -29,42 +30,45 @@ public abstract class TasklistEntity<T extends TasklistEntity<T>> extends Abstra
...
-      final String id, final long key, final String tenantId, final int partitionId) {
+      final String id, final Long key, final String tenantId, final int partitionId) {
     this.id = id;
     this.key = key;
     this.tenantId = tenantId;
     this.partitionId = partitionId;
   }
 
-  public TasklistEntity(final String id, final long key, final String tenantId) {
+  public TasklistEntity(final String id, final Long key, final String tenantId) {
```
Comment (Zelldon) [inline webapps-schema/src/main/java/io/camunda/webapps/schema/entities/tasklist/TasklistEntity.java:40]: 🤔 Shouldn't we delete this entity? Or do we keep the entity but not having the respective index?

Comment (Zelldon) [reply webapps-schema/src/main/java/io/camunda/webapps/schema/entities/tasklist/TasklistEntity.java:40]: Ah can we maybe merge it with the TaskEntity?

File: webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      "completionTime": {
+        "format": "date_time || epoch_millis",
+        "type": "date"
+      },
+      "processInstanceId": {
+        "type": "keyword"
+      },
+      "id": {
+        "type": "keyword"
+      },
```
Comment (Zelldon) [inline webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json:36]: ❓ In Tasklist-list-view there was a taskId. I guess we don't need this?

File: webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      "fullValue": {
+        "type": "keyword",
+        "ignore_above": 8191
+      },
+      "scopeKey": {
+        "type": "keyword"
+      },
+      "isTruncated": {
+        "type": "boolean"
+      }
```
Comment (Zelldon) [inline webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json:120]: ❓ What about 

```

      "isPreview": {
        "type": "boolean"
      },
      "dataType": {
        "type": "keyword"
      }
```

**Review by ChrisKujawa — APPROVED** [review]
In general looks good to me. I had some questions/comments. I guess someone more familiar with the structure should review this as well.

File: webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json (hunk 3 above)
Comment (panagiotisgts) [reply webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json:120]: The `isPreview` field is being replaced by `isTruncated` for consistency with the new REST API format.
The `dataType` will not be needed.

File: webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json (hunk 2 above)
Comment (panagiotisgts) [reply webapps-schema/src/main/resources/schema/elasticsearch/create/template/tasklist-task.json:36]: The `taskId` field has the same value as the `flowNodeBpmnId` which is the Activity ID, so we'll keep only one.

File: webapps-schema/src/main/java/io/camunda/webapps/schema/entities/tasklist/TasklistEntity.java (hunk 1 above)
Comment (panagiotisgts) [reply webapps-schema/src/main/java/io/camunda/webapps/schema/entities/tasklist/TasklistEntity.java:40]: `TasklistEntity` serves as a base class for all entities related to previous Tasklist indices, it contains the three parameters that are common to all entities (id, key and partitionId). We could create a base class for both Operate and Tasklist entities and use as a baseline in later iterations.

**Pull request merged**